src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -49,6 +49,210 @@ namespace AppInstaller::Repository
             bool IsTombstone = false;
         };
 
+        namespace
+        {
+            // Checks whether a default source is enabled with the current settings.
+            // onlyExplicit determines whether we consider the not-configured state to be enabled or not.
+            bool IsDefaultSourceEnabled(std::string_view sourceToLog, ExperimentalFeature::Feature feature, bool onlyExplicit, TogglePolicy::Policy policy)
+            {
+                if (!ExperimentalFeature::IsEnabled(feature))
+                {
+                    // No need to log here
+                    return false;
+                }
+
+                if (onlyExplicit)
+                {
+                    // No need to log here
+                    return GroupPolicies().GetState(policy) == PolicyState::Enabled;
+                }
+
+                if (!GroupPolicies().IsEnabled(policy))
+                {
+                    AICLI_LOG(Repo, Info, << "The default source " << sourceToLog << " is disabled due to Group Policy");
+                    return false;
+                }
+
+                return true;
+            }
+
+            bool IsWingetCommunityDefaultSourceEnabled(bool onlyExplicit = false)
+            {
+                return IsDefaultSourceEnabled(s_Source_WingetCommunityDefault_Name, ExperimentalFeature::Feature::None, onlyExplicit, TogglePolicy::Policy::DefaultSource);
+            }
+
+            bool IsWingetMSStoreDefaultSourceEnabled(bool onlyExplicit = false)
+            {
+                return IsDefaultSourceEnabled(s_Source_WingetMSStoreDefault_Name, ExperimentalFeature::Feature::ExperimentalMSStore, onlyExplicit, TogglePolicy::Policy::MSStoreSource);
+            }
+
+            template<ValuePolicy P>
+            std::optional<SourceFromPolicy> FindSourceInPolicy(std::string_view name, std::string_view type, std::string_view arg)
+            {
+                auto sourcesOpt = GroupPolicies().GetValueRef<P>();
+                if (!sourcesOpt.has_value())
+                {
+                    return std::nullopt;
+                }
+
+                const auto& sources = sourcesOpt->get();
+                auto source = std::find_if(
+                    sources.begin(),
+                    sources.end(),
+                    [&](const SourceFromPolicy& policySource)
+                    {
+                        return Utility::ICUCaseInsensitiveEquals(name, policySource.Name) && Utility::ICUCaseInsensitiveEquals(type, policySource.Type) && arg == policySource.Arg;

```

Comment:
>Utility::ICUCaseInsensitiveEquals(name, policySource.Name) [](start = 31, length = 58)

Should we really be requiring the name to match?  That seems restrictive.

Reply:
For allowed sources, it may be restrictive but it may also be what the IT admin expects: that all the machines that have the policy and want to use the source will use the same name. E.g. so they can think in terms of "is the source foo configured?" instead of "is the REST source from foo.example.com configured?"

Reply:
True, and I suppose if they have a list for the employee of available sources they would provide the name.